app/apptesting/test_suite.go

```diff
@@ -0,0 +1,46 @@
+package apptesting
+
+import (
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/regen-network/regen-ledger/v4/app"
+)
+
+type KeeperTestHelper struct {
+	suite.Suite
+
+	App         *app.RegenApp
+	Ctx         sdk.Context
+	QueryHelper *baseapp.QueryServiceTestHelper
+	TestAccs    []sdk.AccAddress
+}
+
+// Setup sets up basic environment for suite (App, Ctx, and test accounts)
+func (s *KeeperTestHelper) Setup() {
+	s.App = app.NewAppWithCustomOptions(s.T(), false, app.DefaultOptions())
+	s.Ctx = s.App.BaseApp.NewContext(false, tmtypes.Header{Height: 1, ChainID: "regen-1", Time: time.Now().UTC()})
+	s.QueryHelper = &baseapp.QueryServiceTestHelper{
+		GRPCQueryRouter: s.App.GRPCQueryRouter(),
+		Ctx:             s.Ctx,
+	}
+
+	s.TestAccs = CreateRandomAccounts(3)
+}
+
+// CreateRandomAccounts is a function return a list of randomly generated AccAddresses
+func CreateRandomAccounts(numAccts int) []sdk.AccAddress {
+	testAddrs := make([]sdk.AccAddress, numAccts)
+	for i := 0; i < numAccts; i++ {
+		pk := ed25519.GenPrivKey().PubKey()
+		testAddrs[i] = sdk.AccAddress(pk.Address())
+	}
+
+	return testAddrs
+}
```

Comment:
This should probably go in a separate file with the other "test helpers". I think we should migrate `test_helpers.go` to a `testutil` package and include this with the other methods that are not directly related to the app test suite.